src/neo/Ledger/Blockchain.cs

```diff
@@ -133,13 +133,14 @@ public SnapshotCache GetSnapshot()
 
         private void OnImport(IEnumerable<Block> blocks, bool verify)
         {
-            uint currentHeight = NativeContract.Ledger.CurrentIndex(View);
+            DataCache snapshot = View;
+            uint currentHeight = NativeContract.Ledger.CurrentIndex(snapshot);
             foreach (Block block in blocks)
             {
                 if (block.Index <= currentHeight) continue;
                 if (block.Index != currentHeight + 1)
                     throw new InvalidOperationException();
-                if (verify && !block.Verify(View))
+                if (verify && !block.Verify(snapshot))
```

Comment:
You can use the single `snapshot` to verify all the blocks. Otherwise the memory will be full.